src/scaffold/ethereum.test.js

```diff
@@ -243,6 +243,265 @@ export function handleExampleEvent1(event: ExampleEvent1Event): void {
+import { ethereum, BigInt, Bytes } from \"@graphprotocol/graph-ts\"
+import { ExampleEvent, ExampleEvent1 } from \"../generated/Contract/Contract\"
+
+export function createExampleEventEvent(
+  a: BigInt,
+  b: Array<Bytes>,
+  param2: string,
+  c: ethereum.Tuple,
+  d: string
+): ExampleEvent {
+  let exampleEventEvent = changetype<ExampleEvent>(newMockEvent())
+
+  exampleEventEvent.parameters = new Array()
+
+  exampleEventEvent.parameters.push(
+    new ethereum.EventParam(\"a\", ethereum.Value.fromUnsignedBigInt(a))
+  )
+  exampleEventEvent.parameters.push(
+    new ethereum.EventParam(\"b\", ethereum.Value.fromBytesArray(b))
+  )
+  exampleEventEvent.parameters.push(
+    new ethereum.EventParam(\"param2\", ethereum.Value.fromString(param2))
+  )
+  exampleEventEvent.parameters.push(
+    new ethereum.EventParam(\"c\", ethereum.Value.fromTuple(c))
+  )
+  exampleEventEvent.parameters.push(
+    new ethereum.EventParam(\"d\", ethereum.Value.fromString(d))
+  )
+
+  return exampleEventEvent
+}
+
+export function createExampleEvent1Event(a: Bytes): ExampleEvent1 {
+  let exampleEvent1Event = changetype<ExampleEvent1>(newMockEvent())
+
+  exampleEvent1Event.parameters = new Array()
+
+  exampleEvent1Event.parameters.push(
+    new ethereum.EventParam(\"a\", ethereum.Value.fromFixedBytes(a))
+  )
+
+  return exampleEvent1Event
+}
+`)
+  })
+
+  test('Test Files (for indexing events)', () => {
+    const files = scaffoldWithIndexEvents.generateTests()
+    const testFile = files['contract.test.ts']
+    const utilsFile = files['contract-utils.ts']
+
+    expect(testFile).toEqual(`\
+import {
+  assert,
+  describe,
+  test,
+  clearStore,
+  beforeAll,
+  afterAll
+} from \"matchstick-as/assembly/index\"
+import { BigInt, Bytes } from \"@graphprotocol/graph-ts\"
+import { ExampleEvent } from \"../generated/schema\"
+import { ExampleEvent as ExampleEventEvent } from \"../generated/Contract/Contract\"
+import { handleExampleEvent } from \"../src/contract\"
+import { createExampleEventEvent } from \"./contract-utils\"
+
+// Tests structure (matchstick-as >=0.5.0)
+// https://thegraph.com/docs/en/developer/matchstick/#tests-structure-0-5-0
+
+describe(\"Describe entity assertions\", () => {
+  beforeAll(() => {
+    let a = BigInt.fromI32(234)
+    let b = [Bytes.fromI32(1234567890)]
+    let param2 = \"Example string value\"
+    let c = \"ethereum.Tuple Not implemented\"
+    let d = \"Example string value\"
+    let newExampleEventEvent = createExampleEventEvent(a, b, param2, c, d)
+    handleExampleEvent(newExampleEventEvent)
+  })
+
+  afterAll(() => {
+    clearStore()
+  })
+
+  // For more test scenarios, see:
+  // https://thegraph.com/docs/en/developer/matchstick/#write-a-unit-test
+
+  test(\"ExampleEvent created and stored\", () => {
+    assert.entityCount(\"ExampleEvent\", 1)
+
+    // 0xa16081f360e3847006db660bae1c6d1b2e17ec2a is the default address used in newMockEvent() function
+    assert.fieldEquals(
```

Comment:
Maybe a dumb question - but why use the "stringly typed" `fieldEquals`, as opposed to doing

```typescript
const event = ExampleEvent.load("0xa16081f360e3847006db660bae1c6d1b2e17ec2a-1");
assert.stringEquals("Example string value", event.param2);

// etc...
```

Reply:
You don't need to load the entities with `fieldEquals`. Also after loading the entity you'll have to make sure that it is not `null` before trying to access the `param2` field, while fieldEquals also handles that case.

With your example you'll have to do this:

```
const event = ExampleEvent.load("0xa16081f360e3847006db660bae1c6d1b2e17ec2a-1");
if(event) {
  assert.stringEquals("Example string value", event.param2);
}
```

Reply:
True, but then I don't need to retype `"0xa16081f360e3847006db660bae1c6d1b2e17ec2a-1"` and `"ExampleEvent"` over and over again. You also get autocompletion and typechecking on the parameters. 

If you misspell `"param2"` as `"parm2"` you'll get a test failure as opposed to a compilation error.

```
    assert.fieldEquals(
      "ExampleEvent",
      "0xa16081f360e3847006db660bae1c6d1b2e17ec2a-1",
      "parm2",
      "Example string value"
    )
```